Doctrine/Orm/DataProvider.php

```diff
@@ -98,31 +76,19 @@ public function getCollection(ResourceInterface $resource, Request $request)
 
         $manager = $this->managerRegistry->getManagerForClass($resource->getEntityClass());
         $repository = $manager->getRepository($entityClass);
+        $queryBuilder = $repository->createQueryBuilder('o');
 
-        $page = (int) $request->get($this->pageParameter, 1);
+        foreach ($this->extensions as $extension) {
+            $extension->apply($resource, $request, $queryBuilder);
 
-        $itemsPerPage = $this->itemsPerPage;
-        if ($this->enableClientRequestItemsPerPage && $requestedItemsPerPage = $request->get($this->itemsPerPageParameter)) {
-            $itemsPerPage = (int) $requestedItemsPerPage;
-        }
-
-        $queryBuilder = $repository
-            ->createQueryBuilder('o')
-            ->setFirstResult(($page - 1) * $itemsPerPage)
-            ->setMaxResults($itemsPerPage)
-        ;
-
-        foreach ($resource->getFilters() as $filter) {
-            if ($filter instanceof FilterInterface) {
-                $filter->apply($resource, $queryBuilder, $request);
+            if ($extension instanceof QueryResultExtensionInterface) {
+                if ($extension->supportsResult($resource, $request)) {
+                    return $extension->getResult($queryBuilder);
```

Comment:
We should ensure, that the extension is the only one registered `QueryResultExtensionInterface` and is the last called.